### PR TITLE
#3792 Leverage framework for dependencies

### LINKED ISSUE #3792: Leverage dependencies available from framework
**Describe the problem you are trying to solve.**
I do not want STJ listed as a dependency for net 6+. System.Net.Http.Json should be checked if it can also be externalised

**Available workarounds**
Accept un-necessary dependency

**Additional context**
N/a

### PR DESCRIPTION
**Pull request by thompson-tomo**
**What kind of change does this PR introduce?**
update

**What is the current behavior?**
Packages are the same for all TFM's

**What is the new behavior?**
Less dependencies are included for net 6+ due to them being available as part of the framework

**What might this PR break?**
n/a

**Other information**:
closes #3792

### CODE REVIEW

**Review by ChrisPulman — CHANGES_REQUESTED** [review]
As you are right about the inclusion in some versions of Net Core I want to give you this PR however the inclusion only happened with Net Core 7+
Please change the ReactiveUI.csproj file to be as per the following, then I will approve it:
```
﻿<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>$(BaseTargetFrameworks);$(MobileTargetFrameworks)</TargetFrameworks>
    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);$(WindowsTargetFrameworks)</TargetFrameworks>
    <AssemblyName>ReactiveUI</AssemblyName>
    <RootNamespace>ReactiveUI</RootNamespace>
    <PackageDescription>A MVVM framework that integrates with the Reactive Extensions for .NET to create elegant, testable User Interfaces that run on any mobile or desktop platform. This is the base package with the base platform implementations</PackageDescription>
    <PackageId>ReactiveUI</PackageId>
    <!-- Disable "BinaryFormatter is obsolete" warnings for entire project -->
    <NoWarn>$(NoWarn);SYSLIB0011;IDE1006</NoWarn>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">14.2</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">14.0</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
    <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'tizen'">6.5</SupportedOSPlatformVersion>
  </PropertyGroup>
  <ItemGroup>
    <Compile Remove="Platforms\**\*.cs" />
    <Compile Remove="Resources\**" />
    <None Include="Platforms\**\*.cs" />
    <EmbeddedResource Remove="Resources\**" />
    <None Remove="Resources\**" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.StartsWith('netstandard'))">
    <Compile Include="Platforms\netstandard2.0\**\*.cs" />
    <Compile Include="Platforms\shared\**\*.cs" />
    <PackageReference Include="System.ComponentModel" />
    <PackageReference Include="System.ComponentModel.Annotations" />
    <PackageReference Include="System.Diagnostics.Contracts" />
    <PackageReference Include="System.Dynamic.Runtime" />
    <PackageReference Include="System.Runtime.Serialization.Primitives" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.EndsWith('-ios')) or $(TargetFramework.EndsWith('-maccatalyst'))">
    <Compile Include="Platforms\apple-common\**\*.cs" />
    <Compile Include="Platforms\ios\**\*.cs" />
    <Compile Include="Platforms\uikit-common\**\*.cs" />
    <Compile Include="Platforms\mobile-common\**\*.cs" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.EndsWith('-macos'))">
    <Compile Include="Platforms\apple-common\**\*.cs" />
    <Compile Include="Platforms\mac\**\*.cs" />
    <Compile Include="Platforms\mobile-common\**\*.cs" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.EndsWith('-tvos'))">
    <Compile Include="Platforms\apple-common\**\*.cs" />
    <Compile Include="Platforms\tvos\**\*.cs" />
    <Compile Include="Platforms\uikit-common\**\*.cs" />
    <Compile Include="Platforms\mobile-common\**\*.cs" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.EndsWith('-android'))">
    <Compile Include="Platforms\android\**\*.cs" />
    <Compile Include="Platforms\mobile-common\**\*.cs" />
  </ItemGroup>
  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) or '$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net7.0' or '$(TargetFramework)' == 'net8.0' or $(TargetFramework.EndsWith('-windows10.0.17763.0')) or $(TargetFramework.EndsWith('-windows10.0.19041.0')) ">
    <Compile Include="Platforms\net\**\*.cs" />
    <PackageReference Include="System.ComponentModel.Annotations" />
  </ItemGroup>
  <ItemGroup Condition="$(TargetFramework.StartsWith('netstandard')) or $(TargetFramework.StartsWith('net4')) or $(TargetFramework.StartsWith('net6'))">
    <PackageReference Include="System.Text.Json" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Splat" />
    <PackageReference Include="DynamicData" />
  </ItemGroup>
  <ItemGroup>
    <None Update="VariadicTemplates.tt" Generator="TextTemplatingFileGenerator" LastGenOutput="VariadicTemplates.cs" />
    <Compile Update="VariadicTemplates.cs" DesignTime="True" AutoGen="true" DependentUpon="VariadicTemplates.tt" />
  </ItemGroup>
</Project>
```

**Comment by thompson-tomo** [comment]
@ChrisPulman change is done.

File: src/ReactiveUI/ReactiveUI.csproj [hunk 1]
```diff
@@ -54,12 +54,17 @@
   </ItemGroup>
   <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) or '$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net7.0' or '$(TargetFramework)' == 'net8.0' or $(TargetFramework.EndsWith('-windows10.0.17763.0')) or $(TargetFramework.EndsWith('-windows10.0.19041.0')) ">
     <Compile Include="Platforms\net\**\*.cs" />
+  </ItemGroup>
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) or '$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="System.ComponentModel.Annotations" />
+    <PackageReference Include="System.Text.Json" />
```
Comment (dpvreony) [inline src/ReactiveUI/ReactiveUI.csproj:60]: don't need this packagereference as covered on L63

**Review by dpvreony — CHANGES_REQUESTED** [review]

File: src/ReactiveUI/ReactiveUI.csproj (hunk 1 above)
Comment (thompson-tomo) [reply src/ReactiveUI/ReactiveUI.csproj:60]: have removed line 63 so that it is optimised as possible

**Comment by ChrisPulman** [comment]
Hi James, I had thought I had made this simple by giving you the code to fix the issue, I want to give you credit for this change, but the code must build to accept it, before checking changes in please ensure that the solution builds and all tests pass

**Comment by thompson-tomo** [comment]
@dpvreony i am not sure what happened but when i copied the code provided it for some reason didn't like the first line even though nothing was visibly different. As such i have pushed a fresh branch which is compiling on my local machine.

**Review by ChrisPulman — APPROVED** [review]

**Comment by database64128** [comment]
This seems to have also fixed dotnet/wpf#8943 for me.

**Pull request merged**